### PR TITLE
Adapting tabular SimpleModelComparison API

### LINKED ISSUE #1640: [BUG] SingleModelComparison - align and improve API and implementation for computer vision and tabular
Request from the community (Gilles)

**Describe the bug**
Currently, CV and tabular have different param names (e.g. strategy vs. simple_model_type), 
and also different confusing values - e.g. in tabular 'random' is unclear, it's actual 'prior'.
Vision API is preferred. seems it's based on: [scikit-learn's](https://scikit-learn.org/stable/modules/generated/sklearn.dummy.DummyClassifier.html#sklearn-dummy-dummyclassifier).

Additional emphasis:
- Verify the difference between "prior" and "stratified" is clear (if there is a difference. If not then remove redundancy).
- Tabular implementation should contain the relevant strategies (e.g. if uniform and any other are missing).

**Expected behavior**
Both should have a similar signature and option names, like the Computer vision one (param names and options).
Including implementation for required scenarios (if anything needs implementing).
Adding deprecation warnings for previous.

**Links and screenshots**
(Compuer Vision API)[https://docs.deepchecks.com/dev/api/generated/deepchecks.vision.checks.model_evaluation.SimpleModelComparison.html#simplemodelcomparison)
image

(Tabular API)[https://docs.deepchecks.com/dev/api/generated/deepchecks.tabular.checks.model_evaluation.SimpleModelComparison.html#simplemodelcomparison]
image

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

Closes #1640 

A few points to consider:
- What I've done in the PR is simply renaming the values and adding deprecation warnings. I wanted to include the prior and stratified dummy models but it would make this check really complex since they are not for regression.
- For regression - I think a lot is missing. Take a look [here](https://scikit-learn.org/stable/modules/generated/sklearn.dummy.DummyRegressor.html#sklearn.dummy.DummyRegressor).
- Currently, the tabular simple model comparison is too general in my opinion, and I think we should separate it into classification and regression types. This will allow us much more features and different valuable simple dummy models.
- When reading the difference between prior and stratified in the vision check, it was pretty clear to me. Couldn't find better wording to make it clearer. I'm open to suggestions :)

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py [hunk 1]
```diff
@@ -31,18 +32,31 @@
...
     ----------
-    simple_model_type : str , default: constant
-        Type of the simple model ['random', 'constant', 'tree'].
-        + random - select one of the labels by random.
-        + constant - in regression is mean value, in classification the most common value.
-        + tree - runs a simple decision tree.
-    alternative_scorers : Dict[str, Callable], default None
+    strategy : str, default: 'constant'
+        Strategy to use to generate the predictions of the simple model ['uniform', 'most_frequent', 'tree'].
+        * `uniform`: select one of the labels by random. (Previously 'random')
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:54]: If i understood correctly, this is NOT random. it selects a value at random based on the labels distribution.
Should be a different name.
In addition would suggest adding a true random - selects at random from labels.unique_values()

Comment (Nadav-Barak) [reply deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:54]: think it should be done now to avoid additional api changes in future
Same in vision

**Review by Nadav-Barak — COMMENTED** [review]
Agree with your comments, check out https://github.com/deepchecks/deepchecks/issues/1572
and see what else can be added from what you saw

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py (hunk 1 above)
Comment (ItayGabbay) [reply deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:54]: Hmmm, I was misled by the docs. You're right

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py [hunk 2]
```diff
@@ -26,23 +27,40 @@
...
-        Type of the simple model ['random', 'constant', 'tree'].
-        + random - select one of the labels by random.
-        + constant - in regression is mean value, in classification the most common value.
-        + tree - runs a simple decision tree.
-    alternative_scorers : Dict[str, Callable], default None
+    strategy : str, default: 'constant'
+        Strategy to use to generate the predictions of the simple model ['stratified', 'uniform',
+        'most_frequent', 'tree'].
+
+        * `stratified`: select one of the labels by random. (Previously 'random')
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:57]:
```suggestion
        * `stratified`: randomly draw a label based on the train set label distribution. (Previously 'random')
```

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py [hunk 3]
```diff
@@ -257,21 +298,22 @@ def _create_simple_model(self, train_ds: Dataset, task_type: TaskType):
...
         ------
         NotImplementedError
-            If the simple_model_type is not supported
+            If the strategy is not supported
         """
         np.random.seed(self.random_state)
 
-        if self.simple_model_type == 'random':
+        if self.strategy == 'uniform':
+            simple_model = UniformModel()
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:306]: Suggestion: If regression can randomly select from [min,max] instead of available labels

**Review by Nadav-Barak — APPROVED** [review]
Approved with comments

File: deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py [hunk 4]
```diff
@@ -26,23 +27,40 @@
...
-        + constant - in regression is mean value, in classification the most common value.
-        + tree - runs a simple decision tree.
-    alternative_scorers : Dict[str, Callable], default None
+    strategy : str, default: 'constant'
+        Strategy to use to generate the predictions of the simple model ['stratified', 'uniform',
+        'most_frequent', 'tree'].
+
+        * `stratified`: select one of the labels by random. (Previously 'random')
+        * `uniform`: draws predictions uniformly at random from the list of values in y.
+        * `most_frequent`: in regression is mean value, in classification the most common value. (Previously 'constant')
```
[the remark this answers is not in the document]
Comment (ItayGabbay) [reply deepchecks/tabular/checks/model_evaluation/simple_model_comparison.py:59]: I wanted to have the same API for both vision and tabular (and sklearn).
Either way for regression I hope we will split soon.

**Review by Nadav-Barak — APPROVED** [review]

File: deepchecks/utils/simple_models.py [hunk 5]
```diff
@@ -65,3 +65,43 @@ def predict_proba(self, X):  # pylint: disable=invalid-name
         classes = sorted(self.labels.unique().tolist())
         predictions = self.predict(X)
         return create_proba_result(predictions, classes)
+
+
+class ClassificationUniformModel:
```
Comment (JKL98ISR) [inline deepchecks/utils/simple_models.py:70]: is it used in vision too?
if not then any of the util used in vision?

**Review by matanper — APPROVED** [review]
